unused variable in PlaybackStream.cpp

### Summary

When LIBMEDIA_AUDIO_BACKEND is not defined, build fails because of an unused variable in PlaybackStream.cpp

I think the problem was introduced with:
* https://github.com/LadybirdBrowser/ladybird/pull/10823


### Operating system

Linux

### Steps to reproduce

1. Fairly freshly installed Ubuntu 26.04
2. sudo apt install autoconf autoconf-archive automake build-essential ccache cmake curl fonts-liberation2 git glslang-tools libdrm-dev libgl1-mesa-dev libncurses-dev libtool nasm ninja-build pkg-config python3-venv qt6-base-dev qt6-tools-dev-tools qt6-wayland tar unzip zip
3. ./Meta/ladybird.py run

### Expected behavior

Build and run ladybird

### Actual behavior

```
$ ./Meta/ladybird.py run
ninja: Entering directory `/home/jon/src/github.com/LadybirdBrowser/ladybird/Build/release'
[1/13] Building CXX object Libraries/LibMedia/CMakeFiles/LibMedia.dir/Audio/PlaybackStream.cpp.o
FAILED: [code=1] Libraries/LibMedia/CMakeFiles/LibMedia.dir/Audio/PlaybackStream.cpp.o
/usr/bin/ccache /usr/bin/c++ -DENABLE_COMPILETIME_FORMAT_CHECK -DLibMedia_EXPORTS -DUSE_FONTCONFIG=1 -DUSE_VULKAN=1 -DUSE_VULKAN_DMABUF_IMAGES=1 -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=3 -I/home/jon/src/github.com/LadybirdBrowser/ladybird -I/home/jon/src/github.com/LadybirdBrowser/ladybird/Libraries -I/home/jon/src/github.com/LadybirdBrowser/ladybird/Services -I/home/jon/src/github.com/LadybirdBrowser/ladybird/Build/release -I/home/jon/src/github.com/LadybirdBrowser/ladybird/Build/release/Libraries -I/home/jon/src/github.com/LadybirdBrowser/ladybird/Build/release/Services -I/usr/include/libdrm -isystem /home/jon/src/github.com/LadybirdBrowser/ladybird/Build/release/vcpkg_installed/x64-linux-dynamic/include -isystem /home/jon/src/github.com/LadybirdBrowser/ladybird/Build/release/vcpkg_installed/x64-linux-dynamic/lib/pkgconfig/../../include -O2 -g -DNDEBUG -std=c++23 -fPIC -fdiagnostics-color=always -march=native -Wcast-qual -Wformat=2 -Wimplicit-fallthrough -Wlogical-op -Wmissing-declarations -Wmissing-field-initializers -Wsuggest-override -Wno-expansion-to-defined -Wno-invalid-offsetof -Wno-maybe-uninitialized -Wno-shorten-64-to-32 -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wno-user-defined-literals -Werror -Wall -Wextra -fno-exceptions -ffp-contract=off -fstrict-flex-arrays=2 -fstack-protector-strong -fsigned-char -ftrivial-auto-var-init=zero -ggnu-pubnames -fno-semantic-interposition -fvisibility-inlines-hidden -Wno-literal-suffix -Wno-unqualified-std-cast-call -Wvla -Wno-array-bounds -Wno-stringop-overflow -Wno-dangling-reference -fstack-clash-protection -O2 -g1 -fvisibility=hidden -MD -MT Libraries/LibMedia/CMakeFiles/LibMedia.dir/Audio/PlaybackStream.cpp.o -MF Libraries/LibMedia/CMakeFiles/LibMedia.dir/Audio/PlaybackStream.cpp.o.d -o Libraries/LibMedia/CMakeFiles/LibMedia.dir/Audio/PlaybackStream.cpp.o -c /home/jon/src/github.com/LadybirdBrowser/ladybird/Libraries/LibMedia/Audio/PlaybackStream.cpp
/home/jon/src/github.com/LadybirdBrowser/ladybird/Libraries/LibMedia/Audio/PlaybackStream.cpp: In static member function ‘static AK::NonnullRefPtr<Core::Promise<AK::NonnullRefPtr<Audio::PlaybackStream> > > Audio::PlaybackStream::create_platform_or_null(Audio::OutputState, u32, AudioDataRequestCallback, AudioDataRequestCallback)’:
/home/jon/src/github.com/LadybirdBrowser/ladybird/Libraries/LibMedia/Audio/PlaybackStream.cpp:11:168: error: unused parameter ‘platform_data_request_callback’ [-Werror=unused-parameter]
   11 | NonnullRefPtr<PlaybackStream::CreatePromise> PlaybackStream::create_platform_or_null(OutputState initial_output_state, u32 target_latency_ms, AudioDataRequestCallback platform_data_request_callback, AudioDataRequestCallback fallback_data_request_callback)
      |                                                                                                                                               ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-unqualified-std-cast-call’ may have been intended to silence earlier diagnostics
cc1plus: note: unrecognized command-line option ‘-Wno-user-defined-literals’ may have been intended to silence earlier diagnostics
cc1plus: note: unrecognized command-line option ‘-Wno-unused-command-line-argument’ may have been intended to silence earlier diagnostics
cc1plus: note: unrecognized command-line option ‘-Wno-unknown-warning-option’ may have been intended to silence earlier diagnostics
cc1plus: note: unrecognized command-line option ‘-Wno-shorten-64-to-32’ may have been intended to silence earlier diagnostics
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```


### URL for a reduced test case

N/A

### HTML/SVG/etc. source for a reduced test case

```html
N/A
```

### Log output and (if possible) backtrace


```shell
N/A
```

### Screenshots or screen recordings

_No response_

### Build flags or config settings

_No response_

--- a/Libraries/LibMedia/Audio/PlaybackStream.cpp
+++ b/Libraries/LibMedia/Audio/PlaybackStream.cpp
@@ -8,7 +8,7 @@
 
 namespace Audio {
 
-NonnullRefPtr<PlaybackStream::CreatePromise> PlaybackStream::create_platform_or_null(OutputState initial_output_state, u32 target_latency_ms, AudioDataRequestCallback platform_data_request_callback, AudioDataRequestCallback fallback_data_request_callback)
+NonnullRefPtr<PlaybackStream::CreatePromise> PlaybackStream::create_platform_or_null(OutputState initial_output_state, u32 target_latency_ms, [[maybe_unused]] AudioDataRequestCallback platform_data_request_callback, AudioDataRequestCallback fallback_data_request_callback)
 {
     auto promise = PlaybackStream::CreatePromise::construct();
 

